akka-stream/src/main/scala/akka/stream/io/InputStreamSource.scala

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.io
+
+import java.io.InputStream
+
+import akka.stream.io.impl.InputStreamSource
+import akka.stream.scaladsl.Source
+import akka.stream.scaladsl.Source._
+import akka.stream.{ OperationAttributes, javadsl }
+import akka.util.ByteString
+
+import scala.concurrent.Future
+
+object InputStreamSource {
+
+  final val DefaultChunkSize = 2048
```

Comment:
8192 is the maximum size where the implementation of java.io will still try to allocate the intermediate buffer on the stack. For bigger sizes it will use malloc:

http://hg.openjdk.java.net/jdk7/jdk7/jdk/file/9b8c96f96a0f/src/share/native/java/io/io_util.c#l96

I guess that's why BufferInputStream also uses this size. FileChannel, however, seems to use another call path.